cypress/integration/aragon.spec.js

```diff
@@ -11,144 +18,368 @@ context('Aragon', () => {
+        await financeContract.methods.deposit(tokenAddress, value.toString(), 'E2E test deposit')
+        .send( {from: root, gas: 700000}, (error, txHash) => {
+          // it returns tx hash because sending tx
+          console.log('tokens transferred: ',txHash)
+        })
+      }
+    })
+    // plug in fresh dev dao address here
+    cy.visit('http://localhost:3000/#/0x87C62dD8FC7eb0302b121BC6e421A88231955Dcb')
   })
 
   context('Address Book', ()=> {
+    before(() => {
+      cy.contains('button','Address Book',{ timeout: 150000 }).wait(1000).click()
+    })
     it('creates Address Entry', () =>{
-      cy.contains('span.item.MenuPanelAppGroup__ButtonItem-sc-1ahx3fh-1','Address Book',{ timeout: 150000 }).wait(1000).click()
-      // This might be a less verbose way to interact with elements inside an
-      // iframe: https://github.com/cypress-io/cypress/issues/136#issuecomment-328100955
-      .get('iframe').iframe().contains('button','New Entity').click().wait(1000)
-      .get('iframe').iframe().find('input[name="name"]').type('Hello World')
-      .get('iframe').iframe().find('input[name="address"]').type('0xb4124ceb3451635dacedd11767f004d8a28c6ee7')
-      .get('iframe').iframe().contains('button','Submit Entity').click().wait(1000)
+      
+      cy.getWithinIframe('button:contains("New entity")').trigger('click')
+      .getWithinIframe('input[name="name"]').type('Hello World', { force: true })
+      .getWithinIframe('input[name="address"]').type('0xb4124ceb3451635dacedd11767f004d8a28c6ee7', { force: true })
+      .getWithinIframe('button:contains("Submit Entity")').click()
 
       cy.contains('button','Create transaction').click()
-      cy.contains('button','Close').click().wait(1000)
-      cy.get('iframe').iframe().contains('div','Hello World')
+      cy.contains('button','Close').click()
+      // a basic validations that the entry exists in the table
+      //cy.getWithinIframe('div.TableView___StyledTd-aczwu3-3:contains("Hello World")')
     })
     it('removes Address Entry', () => {
-      cy.get('iframe').iframe().find('div.ContextMenu__BaseButton-ris724-1').click()
-      .get('iframe').iframe().contains('div','Remove').click().wait(1000)
+      // click the button that opens the context menu
+      cy.getWithinIframe('button.fcvbGI').trigger('click')
+      .getWithinIframe('Button:contains("Remove")').trigger('click').wait(1000)
 
       cy.contains('button','Create transaction').click()
       cy.contains('button','Close').click().wait(1000)
     })
   })
-/*
-  it('.children() - get child DOM elements', () => {
-    // https://on.cypress.io/children
-    cy.get('.traversal-breadcrumb')
-      .children('.active')
-      .should('contain', 'Data')
-  })
 
-  it('.closest() - get closest ancestor DOM element', () => {
-    // https://on.cypress.io/closest
-    cy.get('.traversal-badge')
-      .closest('ul')
-      .should('have.class', 'list-group')
-  })
+  context('Allocations', () => {
```

Comment:
I would like to see many test files, rather than one giant one. Can we put test files near the code where they're used, the way people do with Jest tests? That would be my favorite way to organize them. Failing that, I'd at least like to see a directory per app. 

* `cypress/integration/projects/add-new-github-project.js`
* `cypress/integration/projects/fund-issue.js`
* `cypress/integration/projects/submit-application.js`

Do you think this pattern would work?

Reply:
This pattern would work, but it would be slower, since each new file will require a fresh page load and syncing. I'd prefer to just keep one file so that automated testing can run as quickly as possible, and when we write tests for syncing, we can go this route.

Reply:
Hmm, ok. Maybe we could extract individual bits of this into functions, to keep this file manageable. 

Can we run separate tests in parallel? Test Rewards in one process while we test Projects in another, for example?

Reply:
> This pattern would work, but it would be slower, since each new file will require a fresh page load and syncing. I'd prefer to just keep one file so that automated testing can run as quickly as possible, and when we write tests for syncing, we can go this route.


What is preventing us from using one test file in the root while importing tests from other files there? I think it would be the same result, in terms of quickness, with the added benefit of having a manageable suite, so I would take that route.

Reply:
I've decided to split the files up by app. I already appreciate how much cleaner the devX is with it this way
> Can we run separate tests in parallel? Test Rewards in one process while we test Projects in another, for example?
We could possibly do parallelization. I'll have to play around it with it more because I'm not sure how the web3 provider willl handle nonces if multiple instances of it are commiting transactions simultaneously. We apparently have a partner building a really smart web3 provider for this kind of thing but I haven't played with it at all yet.